src/app/clusters/ota-requestor/ota-requestor-interface.h

```diff
@@ -0,0 +1,40 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/* This file contains the declaration for the OTA Requestor interface.
+ * Any implementation of the OTA Requestor (e.g. the OTARequestor class) must implement
+ * this interface.
+ */
+
+#pragma once
+
+// Interface class to connect the OTA Software Update Requestor cluster command processing
+// with the core OTA Requestor logic. The OTARequestor class implements this interface
+class OTARequestorInterface
+{
+public:
+    // Handler for the AnnounceOTAProvider command
+    virtual bool HandleAnnounceOTAProvider(
+        ip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
+        const chip::app::Clusters::OtaSoftwareUpdateRequestor::Commands::AnnounceOtaProvider::DecodableType & commandData ch) = 0;
+
+    // Handler for the QueryImageResponse command
+    virtual bool HandleQueryImageResponse(void * context, uint8_t status, uint32_t delayedActionTime, CharSpan imageURI,
```

Comment:
I believe this function basically will replace `OnQueryImageResponse` that currently exists. Can you also have this take in a `QueryImageResponse::DecodableType` instead of individual params as it is currently listed?

Reply:
Actually I'm a bit vague on this. What determines the parameter set for the callback that is passed to cluster.InvokeCommand() ?

Reply:
Currently, it is just arbitrary values passed in but we should make them configurable

```
   QueryImage::Type args;
    args.vendorId           = kExampleVendorId;
    args.productId          = kExampleProductId;
    args.softwareVersion    = kExampleSoftwareVersion;
    args.protocolsSupported = kExampleProtocolsSupported;
    args.hardwareVersion.Emplace(kExampleHWVersion);
    args.location.Emplace(exampleLocation);
    args.requestorCanConsent.Emplace(kExampleClientCanConsent);
    args.metadataForProvider.Emplace(metadata);
    err = cluster.InvokeCommand(args, /* context = */ nullptr, OnQueryImageResponse, OnQueryImageFailure);
```

Reply:
Yes. But I do not understand what determines this set of parameters. How does the code executing InvokeCommand() know that the set of parameters passed to the function in its third argument is of the type that OnQueryImageResponse() expects? I don't see in the code any sort of registration of this info.